tests/common/plugins/custom_fixtures/__init__.py

```diff
@@ -6,5 +6,11 @@ def pytest_collection_modifyitems(items):
     for item in items:
         asic_marks = [mark for mark in item.iter_markers(name="asic")]
         if asic_marks:
-            item.fixturenames.append("check_dut_asic_type")
+            for pos, name in enumerate(item.fixturenames):
+                try:
+                    if item._fixtureinfo.name2fixturedefs[name][0].scope == 'module':
+                        break
+                except:
+                    pos = len(item.fixturenames)
+            item.fixturenames.insert(pos, "check_dut_asic_type")
```

Comment:
if this loop executes to completion, then this fixture gets inserted as the second last item. Is that what is intended?

Reply:
Thanks Neetha. It's a bug. I'll post an update.

Reply:
Updated. Please help to re-check. Thanks @neethajohn